src/plugins/service-worker/index.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+const buildPaths = require('mwp-cli/src/commands/buildCommands/config/paths');
+const path = require('path');
+
+/*
+ * The service worker plugin will register a root-level route for serving the
+ * service worker script file.
+ *
+ * Disk location of the script file is determined by builder CLI config import.
+ */
+export default function register(
+	server: Object,
+	options: ?Object,
+	next: () => void
+) {
+	/*
+	 * Route for service worker script at top-level path. Depends on `Inert`
+	 * `path` must match client `serviceWorker.register` call - MWP provides this
+	 * in the `<ServiceWorker>` component
+	 */
+	server.route({
+		method: 'GET',
+		path: '/asset-service-worker.{localeCode}.js',
```

Comment:
the service worker is generated by a plugin that is defined once-per-build, which ends up meaning once-per-language. We actually want to keep that separation, because the way the plugin works is to check the filenames of all chunks emitted by the build, and define them all as 'precache' resources - we wouldn't want a single service worker to try to precache all versions of the app in all languages, which would be many megabytes of assets.